superweather/superweather/ViewController.m

```diff
@@ -0,0 +1,35 @@
+//
+//  ViewController.m
+//  superweather
+//
+//  Created by mikael petersson on 2016-04-17.
+//  Copyright © 2016 mikael petersson. All rights reserved.
+//
+
+#import "ViewController.h"
+
+@interface ViewController()
+
+@property (strong, nonatomic) LocationTool* location;
+@property (weak, nonatomic) IBOutlet UILabel *lattitudeLabel;
+@property (weak, nonatomic) IBOutlet UILabel *longditudeLabel;
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.location = [LocationTool sharedTool];
+    [self.location addListener:self];
```

Comment:
We should be careful to remove the listener also. Usually it's safer to add/remove listeners in init/dealloc